### ISSUE TITLE
Not handling DOCTYPE element

### ISSUE THREAD

**Issue by matthew-green** [chrisbottin/xml-formatter#8]
I've got an xml that includes a DOCTYPE right at the top. It seems that something within the formatter is not handling this element correct and not throwing an error when it does because when I remove the DOCTYPE the xml is formatted without issue. 

Is this a known limitation or something that could be addressed?

**Comment by chrisbottin** [comment]
Hi,

Can you please attach an example of the XML input you are trying to use?

Thanks,

Chris

**Comment by matthew-green** [comment]
I'm not sure how much I can supply but its a log4j.xml config file. Here is a snippet. If you need more I can find out what all I'm allowed to post but hopefully this gives you enough to work with.  

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE log4j:configuration SYSTEM "log4j.dtd">
<log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
  <appender name="foo" class="org.apache.log4j.DailyRollingFileAppender">
    ...
  </appender>
  ...
</log4j:configuration>
```

**Comment by chrisbottin** [comment]
Thanks, the problem relates to the original library parsing the XML. Just checked the library and the same DOCTYPE issue was reported on the original repository https://github.com/segmentio/xml-parser/issues/23.

I have already forked this repository for providing advanced options. I will make a fix on my repository and give you a shout when it's done in couple of days I reckon.

**Comment by chrisbottin** [comment]
This should be fixed by #15
